tools/autograph-monitor/contentsignature_test.go

```diff
@@ -0,0 +1,403 @@
+	}()
+	chain, err := getX5U("http://localhost:64322/wronglyorderedchain")
+	if err != nil {
+		t.Fatalf("Failed to retrieved certificate chain: %v", err)
+	}
+	err = verifyCertChain(chain)
+	if err == nil {
+		t.Fatal("Expected to fail chain verification with cert not signed by parent, but succeeded")
+	}
+	log.Printf("Chain verification failed with: %v", err)
+	if !strings.Contains(err.Error(), "is not signed by parent certificate") {
+		t.Fatalf("Expected to failed with certificate not being signed by parent, but failed with: %v", err)
+	}
+}
+
+func TestVerifyFirefoxRoot(t *testing.T) {
+	conf.RootHash = "97:E8:BA:9C:F1:2F:B3:DE:53:CC:42:A4:E6:57:7E:D6:4D:F4:93:C2:47:B4:14:FE:A0:36:81:8D:38:23:56:0E"
+	block, _ := pem.Decode(FirefoxPKIRootPEM)
+	if block == nil {
+		t.Fatalf("Failed to parse certificate PEM")
+	}
+	certX509, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Fatalf("Could not parse X.509 certificate: %v", err)
+	}
+	err = verifyRoot(certX509)
+	if err != nil {
+		t.Fatalf("Failed to verify valid Firefox root certificate: %v", err)
+	}
+}
+
+func TestVerifyFirefoxRootWithBadHash(t *testing.T) {
+	conf.RootHash = "foo"
+	block, _ := pem.Decode(FirefoxPKIRootPEM)
+	if block == nil {
+		t.Fatalf("Failed to parse certificate PEM")
+	}
+	certX509, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Fatalf("Could not parse X.509 certificate: %v", err)
+	}
+	err = verifyRoot(certX509)
+	if err == nil {
+		t.Fatalf("Expected to fail with incorrect hash, but succeeded.")
+	}
+	if !strings.Contains(err.Error(), "hash does not match expected root") {
+		t.Fatalf("Expected to fail with hash mismatch, but failed with: %v", err)
+	}
+}
+
+func TestVerifyFirefoxStagingRoot(t *testing.T) {
+	conf.RootHash = "DB:74:CE:58:E4:F9:D0:9E:E0:42:36:BE:6C:C5:C4:F6:6A:E7:74:7D:C0:21:42:7A:03:BC:2F:57:0C:8B:9B:90"
+	block, _ := pem.Decode(FirefoxPKIStagingRootPEM)
+	if block == nil {
+		t.Fatalf("Failed to parse certificate PEM")
+	}
+	certX509, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Fatalf("Could not parse X.509 certificate: %v", err)
+	}
+	err = verifyRoot(certX509)
+	if err != nil {
+		t.Fatalf("Failed to verify valid Firefox root certificate: %v", err)
+	}
+}
+
+// fixtures -----------------------------------------------------------------
+
+var ValidMonitoringContentSignature = signatureresponse{
+	Ref:       "1881ks1du39bi26cfmfczu6pf3",
+	Type:      "contentsignature",
+	Mode:      "p384ecdsa",
+	SignerID:  "normankey",
+	PublicKey: "MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAEVEKiCAIkwRg1VFsP8JOYdSF6a3qvgbRPoEK9eTuLbrB6QixozscKR4iWJ8ZOOX6RPCRgFdfVDoZqjFBFNJN9QtRBk0mVtHbnErx64d2vMF0oWencS1hyLW2whgOgOz7p",
+	Signature: "9M26T-1RCEzTAlCzDZk6CkEZxkVZkt-wUJfA4s4altKx3Vw-MfuE08bXy1TenbR0I87PzuuA9c1CNOZ8hzRbVuYvKnOH0z4kIbGzAMWzyOxwRgufaODHpcnSAKv2q3JM",
+	X5U:       "https://raw.githubusercontent.com/mozilla-services/autograph/master/docs/statics/normandy.content-signature.mozilla.org-20210705.dev.chain",
```

Comment:
I should probably use a localhost server to serve this file instead of reaching out to github. What do you think @g-k ?